### ISSUE TITLE
Versions of Proton installed after index is created aren't recognized

### ISSUE THREAD

**Issue by benblank** [caverym/proton-caller#71]
**Describe the bug**

If you install a version of Proton after `proton-call` has built its index, the new version does not appear to ever be picked up.

**To Reproduce**

1. Install `proton-call`.
2. Run `proton-call --index`.
3. Install a version of Proton which was not previously installed, via Steam.
4. Run `proton-call --index` again.

**Expected behavior**

The new version of Proton is listed.

**Screenshots**

Here's a "screenshot" of the behavior in my terminal:

```text
/home/five35 $ proton-call --version
Proton Caller (proton-call) 3.1.0 Copyright (C) 2021 Avery Murray <avery.murray@caverym.xyz>
/home/five35 $ proton-call --index
Indexed Directory: /home/five35/.local/share/Steam/steamapps/common/

Indexed 0 Proton Versions:

/home/five35 $ ls -ld /home/five35/.local/share/Steam/steamapps/common/
drwxrwxr-x 6 five35 five35 4,096 Nov 20 09:47 /home/five35/.local/share/Steam/steamapps/common/
/home/five35 $ ls -l .cache/proton/index 
-rw-rw-r-- 1 five35 five35 65 Nov 20 09:07 .cache/proton/index
/home/five35 $ rm .cache/proton/index 
/home/five35 $ proton-call --index
warning: failed read/write to cache: can't deserialize
reindexing...
Indexed Directory: /home/five35/.local/share/Steam/steamapps/common/

Indexed 1 Proton Versions:

Proton 7.0: /home/five35/.local/share/Steam/steamapps/common/Proton 7.0
/home/five35 $
```

**Desktop**

- OS: Ubuntu Desktop 22.10
- Steam: (native)
  - Built: Nov 17 2022, at 02:45:32
  - Steam API: v020
  - Steam package versions: 1668654564
- Proton: 7.0
  - App ID: 1887720
  - Build ID: 9317711
  - Installed content updated: Nov 20, 2022 at 9:46 AM

**Additional context**

Note that, while I've described the problem in terms of `proton-call --index`, it also prevents e.g. `proton-call --run foo.exe` from using the new version of Proton. In my case, I happened to have from `proton-call` prior to installing *any* version of Proton , which meant `proton-call --run` wouldn't work at all.

Based on the behavior, I suspect versions of Proton which have been uninstalled are similarly never removed from the index.

I'm not sure what the best approach is for making sure the index is up to date. In this case, the last-modified-time of `/.local/share/Steam/steamapps/common` is newer than that of the index file, but I don't know whether that can be relied upon.

**Comment by caverym** [comment]
iirc, proton-caller will reindex if a version asked for isn't found. Try running something with an unindexed version, it should be picked up

**Comment by Mart-Bogdan** [comment]
I think I've fallen into same problem:

```
$ proton-call -r MenuSystem.exe                                 
proton-call: failed to create Proton directory: failed to create Proton directory: No such file or directory (os error 2)

```

**Comment by Mart-Bogdan** [comment]
I have to create /home/winnie/Documents/Proton/env/  dir manually.

It's been a while since I first installed proton caller and it just worked. Perhaps something deleated it.


Is it possible to use profile directory used by steam, not separate on for proton-caller?

**Comment by caverym** [comment]
> I have to create /home/winnie/Documents/Proton/env/ dir manually.

Yeah. I've overlooked having it create this directory forever. It's about time I get around to having it make sure everything in the config exists.

> Is it possible to use profile directory used by steam, not separate on for proton-caller?
Yes, just change the config to what you would like. The default and example configs are just what I used when I originally created Proton Caller. Configs are there for customization :)

**Comment by TheStutterNerd** [comment]
Just had this exact same issue. The thing is, I moved all my steam games/protons to another drive. I changed the config to the new directory but it was only picking up 1 proton version in the old location (which didn't even exist anymore).

I did try runnning a specific version (as mentioned above, which wasn't getting picked up) and it never regenerated a new index within cache. 

The locations was showing the new directory when I ran `proton-caller -i` too, but I had to manually remove the index for it all to work as specified above.
